test/cog/chat/hipchat/template_processor_test.exs

```diff
@@ -107,4 +107,37 @@ defmodule Cog.Chat.HipChat.TemplateProcessorTest do
 
     assert expected == rendered
   end
+
+  test "handles link directives" do
+    directives = [
+      %{"name" => "link", "text" => "a link", "url" => "http://www.example.com"}
+    ]
+
+    rendered = TemplateProcessor.render(directives)
+    expected = "<a href='http://www.example.com'>a link</a>"
+
+    assert expected == rendered
+  end
+
+  test "handles link directive with a nil url" do
```

Comment:
Pretty sure if that happens, it's unintentional, and the user should be made aware of it somehow.